### PR TITLE
Ollama max_tokens settings

### LINKED ISSUE #15200: Requesting ability to set max_tokens for ollama models
### Check for existing issues

- [X] Completed

### Describe the feature

There doesn't seem to be any way to set the `max_tokens` for an ollama assistant model with the 0.147.0 nightly I'm using. I've attempted to set it in the config in a variety of ways and none of them had any visible impact in the UI. I always see "_n_ / 2k" in the top right corner, no matter which model I select.

### If applicable, add mockups / screenshots to help present your vision of the feature

image

### LINKED ISSUE #17012: Zed crashes the ollama runner
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

After #16877 any prompt to ollama with llama3.1:latest crashes the ollama runner (ROCm). This does not happen with `ollama run llama3.1:latest` which uses the default token limit of 2048.

### Environment

Zed: v0.151.0 (Zed Dev a5b82b2bf3b54ec210bb293cf541eb4c6164824b)
OS: Linux Wayland ubuntu 24.04
Memory: 31.3 GiB
Architecture: x86_64
GPU: Radeon Vega Frontier Edition (RADV VEGA10) || radv || Mesa 24.0.9-0ubuntu0.1

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by notpeter**
- Closes https://github.com/zed-industries/zed/issues/15200
- Closes https://github.com/zed-industries/zed/issues/17012
- Clamp default max tokens (context length) to 16384.
- Add documentation for ollama context configuration.

Release Notes:

- Ollama: Improved handling of variable context lengths; support for user configurable `max_tokens` in setting.json.

### CODE REVIEW

File: crates/ollama/src/ollama.rs [hunk 1]
```diff
@@ -109,7 +116,13 @@ impl Model {
     }
 
     pub fn display_name(&self) -> &str {
-        &self.name
+        if let Some(display_name) = &self.display_name {
+            display_name
+        } else if let Some(short) = self.name.strip_suffix(":latest") {
+            short
```
Comment (rgbkrk) [inline crates/ollama/src/ollama.rs:122]: Nice!

File: docs/src/assistant/configuration.md [hunk 2]
```diff
@@ -108,33 +108,49 @@ Custom models will be listed in the model dropdown in the assistant panel.
...
+          "max_tokens": 32768
+        }
+      ]
     }
   }
 }
 ```
 
+If you specify a context length that is too large for your hardware, Ollama will log an error. You can watch these logs by running: `tail -f ~/.ollama/logs/ollama.log` (MacOS) or `journalctl -u ollama -f` (Linux). Depending on the memory available on your machine, you may need to adjust the context length to a smaller value.
+
```
Comment (rgbkrk) [inline docs/src/assistant/configuration.md:153]: Truly solid docs here overall.

File: crates/language_model/src/provider/ollama.rs [hunk 3]
```diff
@@ -123,10 +136,32 @@ impl LanguageModelProvider for OllamaLanguageModelProvider {
...
+            models.insert(
+                model.name.clone(),
+                ollama::Model {
+                    name: model.name.clone(),
+                    display_name: model.display_name.clone(),
+                    max_tokens: model.max_tokens,
+                    keep_alive: None,
+                },
+            );
+        }
```
Comment (rgbkrk) [inline crates/language_model/src/provider/ollama.rs:161]: Settings > API listing 👍

**Review by rgbkrk — APPROVED** [review]

File: crates/language_model/src/provider/ollama.rs (hunk 3 above)
Comment (smm9509) [reply crates/language_model/src/provider/ollama.rs:161]: 👎, the settings seem to be LESS DETAILED THAN the API listing! that means, if a user wants to configure the settings for a model they got from the API, the user's settings are overridden (not merged!) with the capabilities (like tool use) reported by the API. (I might make a bug report issue about this, I'm experiencing this with 
```
{
          "name": "qwen3:4b",
          "supports_thinking": true,
          "max_tokens": 32768,
        },
``` which seems to override the tools from line 142 with a model from line 161 that does not have tools but respects my context size preferences.

**Pull request merged**